### DISCUSSION TITLE
Feature Request: "Please correct me if I'm wrong" for GitHub Copilot

### DISCUSSION THREAD

**Discussion by vifeng** [community/community#128021] [Copilot Conversations]
### Select Topic Area

Product Feedback

### Feature Request: "Please correct me if I'm wrong" for GitHub Copilot

**Description:**

I would like to propose a new feature for GitHub Copilot called "Please correct me if I'm wrong." This feature would emit a warning when a statement or code snippet appears to be factually incorrect. For example, if a user writes "In a singleton each instance will have a different reference," Copilot would detect this as incorrect and suggest the correct statement, such as "In a singleton, there is only one instance, and every reference points to that same instance."

**Use Cases:**

- Writing accurate documentation and comments in code.
- Ensuring presentations and educational materials are factually correct.
- Assisting new developers in learning correct programming concepts.

**Benefits:**

- Improved code quality and correctness.
- Enhanced learning experience for new developers.
- Time-saving by reducing the need for manual fact-checking.

**Implementation:**

The feature could leverage natural language processing and existing AI models to detect inaccuracies and suggest corrections. Additionally, providing links to documentation or resources for further reading would be beneficial.

Thank you for considering this feature. I believe it would be a valuable addition to GitHub Copilot.

**Comment by HaShingThang** [comment]
This feature would warn users about factually incorrect statements or code snippets, improving accuracy in documentation, comments, and educational materials. It benefits code quality, developer learning, and saves time by reducing manual fact-checking. Implementing it with NLP and AI models, along with links to further resources, would make Copilot even more valuable. Thank you for considering this enhancement.
